applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -86,6 +86,80 @@ void NavierStokesWallCondition<2,2>::GetDofList(DofsVectorType& rElementalDofLis
     }
 }
 
+/**
+ * @see NavierStokesWallCondition::Initialize
+ * 
+ * @tparam TDim 
+ * @tparam TNumNodes 
+ */
+
+template<unsigned int TDim, unsigned int TNumNodes>
+void NavierStokesWallCondition<TDim,TNumNodes>::Initialize()
+{
+	KRATOS_TRY;
+	const array_1d<double,3>& rNormal = this->GetValue(NORMAL);
+     // std::cout << rNormal << std::endl;
+	if (norm_2(rNormal) == 0.0){
+	    std::cout << "error on condition -> " << this->Id() << std::endl;
+	    KRATOS_THROW_ERROR(std::logic_error, "NORMAL must be calculated before using this condition","");
+	}
+	if (mInitializeWasPerformed){
+        KRATOS_ERROR_IF(mpParentElement == NULL) << "Parent as NULL pointer (Check 1)";
+	    return;
+	}
+
+	mInitializeWasPerformed = true;
+	// double EdgeLength;
+	array_1d<double,3> Edge;
+	GeometryType& rGeom = this->GetGeometry();
+	WeakPointerVector<Element> ElementCandidates;
+    
+    // std::cout << "before finding candidates" << std::flush;
+	for (SizeType i = 0; i < TNumNodes; i++){
+		WeakPointerVector<Element>& rNodeElementCandidates = rGeom[i].GetValue(NEIGHBOUR_ELEMENTS);
+
+         // Check that the condition has candidate parent elements
+         KRATOS_ERROR_IF(rNodeElementCandidates.size() == 0) <<
+             "Condition " << this->Id() << " has no candidate parent elements.\n" <<
+             "Check that the FindNodalNeighboursProcess has been executed ( e.g in Solver.Initialize() )";
+
+		for (SizeType j = 0; j < rNodeElementCandidates.size(); j++){
+
+            ElementCandidates.push_back(rNodeElementCandidates(j));
+		}
+	}
+
+	std::vector<IndexType> NodeIds(TNumNodes), ElementNodeIds;
+	for (SizeType i=0; i < TNumNodes; i++){
+		NodeIds[i] = rGeom[i].Id();
+	}
+
+	std::sort(NodeIds.begin(), NodeIds.end());
```

Comment:
did you check if this will this cause problems in other places?

Reply:
Thank you for the hint. 
I took closer looks at similar cases and ran some smaller (exemplaric) test cases. From my point, I could not spot any problems. Also, a similar procedure is applied other conditions that need to find their parent element. An example is https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FluidDynamicsApplication/custom_conditions/fs_werner_wengle_wall_condition.h#L240 at line 240.
Could you specify what is your main concern here, please?

Reply:
The discussed variable is used now and abbreviates the neighbour search. Thank you very much for the hint.